### PR TITLE
docs/ami add homepage with ami creation process and details

### PR DESCRIPTION
**Pull request by abeatrix**
Re: https://sourcegraph.slack.com/archives/C02E4HE42BX/p1665684667007539?thread_ts=1665618337.109499&cid=C02E4HE42BX

This PR adds a home page to the machine images deployment method and details about our creation and publication process.

## Test plan

Drafted docs for review

### CODE REVIEW

File: doc/admin/deploy/machine-images/index.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+- Ability to perform upgrades easily with or without SSH access
+
+#### Creation procress
+
+Our AWS AMIs are all based on the HVM method, which provides us with the ability to create a Sourcegraph instance directly on the virtual machine using the verified Amazon Linux 2 Kernel 5.10 x86_64 HVM as the base image. The following steps are performed during the AMI creation process through our [install script](https://sourcegraph.com/github.com/sourcegraph/deploy@v4.0.1/-/blob/install/install.sh):
+1. De-escalate to ec2-user to make sure tasks are performed by non-root user
+1. Install Git
+1. Clone the deployment repository
+1. Data volumes are formatted and labeled for the AMI instance to mount later
+1. Add configurations to the kernel that would otherwise limit Sourcegraph search performance
```
Comment (jjeffwarner) [inline doc/admin/deploy/machine-images/index.md:40]: Can we provide a link to the changes?

File: doc/admin/deploy/machine-images/index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+1. Adjust the local firewall to ensure k3s cluster networking/DNS can pass through
```
Comment (jjeffwarner) [inline doc/admin/deploy/machine-images/index.md:41]: Also a good place to supply a link if available.

File: doc/admin/deploy/machine-images/index.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+- K3s exposes the kubelet API to the Kubernetes control plane node through a websocket tunnel in order to eliminate the need to expose extra ports on the worker nodes.
+- No ingress requirements which allow cluster owners to restrict inbound traffic to only traffic within their network
+- Built-in certificate rotation with the expiration date of 12 months (docs)[https://docs.k3s.io/advanced]
+- Enable custom certificates through [etcdctl](https://docs.k3s.io/advanced#using-etcdctl) as recommended by k3s.
+- Ability to launch an AMI instance on EC2 with custom encryption [docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIEncryption.html#AMI-encryption-launch).
+
+Please refer to the official [CIS hardening guide](https://docs.k3s.io/security/hardening-guide) by k3s for more details and general tips on how to improve security of your cluster.
+
+## Additional resources
```
Comment (christinelovett) [inline doc/admin/deploy/machine-images/index.md:81]: Is this a fair place to link our [vulnerability and patching policy](https://handbook.sourcegraph.com/departments/engineering/dev/policies/vulnerability-management-policy/#vulnerability-management-policy)
and [SLAs](https://handbook.sourcegraph.com/departments/engineering/dev/policies/vulnerability-management-policy/#vulnerability-service-level-agreements)

File: doc/admin/deploy/machine-images/index.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# Sourcegraph Machine Images
+
+We aim to improve the overall deployment experience for our users through the customized machine images.
```
Comment (beyang) [inline doc/admin/deploy/machine-images/index.md:3]:
```suggestion
We aim to improve the overall deployment experience for our users through customized machine images.
```

**Review by beyang — APPROVED** [review]

File: doc/admin/deploy/machine-images/index.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+1. Link the ephemeral kubelet/pod storage to our data disk
+1. Link the persistent volume pod storage to our data disk
+1. Link the k3s's embedded database to our data disk
+1. Install k3s on root volume
+1. Correct permission of the k3s kube config file located in /etc/rancher/k3s/k3s.yaml
+1. Install Helm
+1. Download Sourcegraph Helm Charts
+1. Deploy Sourcegraph using the local Helm Charts
+1. Save the version number to both root and data volumes
+1. Add cronjob to run the [reboot script](https://sourcegraph.com/github.com/sourcegraph/deploy@v4.0.1/-/blob/install/reboot.sh) on each reboot
```
Comment (jdpleiness) [inline doc/admin/deploy/machine-images/index.md:51]:
```suggestion
1. Add a cronjob to run the [reboot script](https://sourcegraph.com/github.com/sourcegraph/deploy@v4.0.1/-/blob/install/reboot.sh) on each reboot
```

**Review by christinelovett — APPROVED** [review]

**Pull request merged**